app/components/layouts/reducer.tsx

```diff
@@ -1,21 +1,29 @@
-import { LayoutState, LAYOUT_INITIAL_STATE, UserDevice } from './records';
-import { ACTION_TYPES } from '../../actions/actionTypes';
+import { createSlice, PayloadAction } from 'redux-starter-kit';
 
-export function reducer(state: LayoutState = LAYOUT_INITIAL_STATE, action: ReduxAction<any>): LayoutState {
-  switch (action.type) {
-    case ACTION_TYPES.SET_DEVICE_TO_DESKTOP: {
-      return { ...state, userDevice: UserDevice.DESKTOP };
-    }
+export enum UserDevice {
+  DESKTOP,
+  TABLET,
+  MOBILE,
+}
 
-    case ACTION_TYPES.SET_DEVICE_TO_TABLET: {
-      return { ...state, userDevice: UserDevice.TABLET };
-    }
+export interface LayoutState {
+  userDevice: UserDevice;
+}
 
-    case ACTION_TYPES.SET_DEVICE_TO_MOBILE: {
-      return { ...state, userDevice: UserDevice.MOBILE };
-    }
+export const LAYOUT_INITIAL_STATE = { userDevice: UserDevice.DESKTOP };
 
-    default:
-      return state;
-  }
-}
+type SetDeviceAction = PayloadAction<{ userDevice: UserDevice }>;
```

Comment:
Can you use this type to type `Dispatch`? How does that work?

Reply:
I think you can't use it directly inside of `Dispatch` type definition. you should make the custom wrapper type definition to cover the strict bond between the component and the action creators.

However, I think we won't need to make the Dispatch definition anymore.
The official doc said that use `export const { setDeviceType } = layoutSlice.actions;` with `connect` and `mapDispatchToProps`.
If you follow the doc, you don't need the definition of action types and dispatch.

If you don't want to use `connect` API and want to handle it with hooks, I think you should make the custom type definition. (I omit it)

Mark(RSK maintainer) said he will update the type definition and use case with hooks API soon.

Reply:
That's true. Since the creators and reducers are created together by the same library, the library guarantees the correctness. So we don't need to type them.